buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -671,8 +671,12 @@ class BeamModulePlugin implements Plugin<Project> {
         include "**/*TestCase.class"
       }
 
-      // Configure all test tasks to use JUnit
-      project.tasks.withType(Test) { useJUnit {} }
+      project.tasks.withType(Test) {
+        // Configure all test tasks to use JUnit
+        useJUnit {}
+        // default maxHeapSize on gradle 5 is 512m, lets increase to handle more demanding tests
+        maxHeapSize = '2g'
```

Comment:
Actually, I started of by adding 'maxHeapSize' to individual projects build.gradle. Have been 3 (or 4?) for javaPreCommit.

I do not have a strong opinion on which to prefer. Of course, doing it explicitly might reveal some issues, where we will have some leak. Downside will be those, why does that test not work here... same test works there...

So I d sa<, as u like...

Reply:
A desirable end state would be to have as many test tasks as possible using the default lower memory. But, we will likely have many today that need higher mem, for not obvious reasons.

This is code debt that would be nice-to-fix at some point, but I think it's best if we decouple it from the Gradle 5 migration. It would be unfortunate if we accidentally destabilized the tests due to this change and the Gradle 5 upgrade got rolled back.

So, my preference would be to file a JIRA to later come back and make this configurable per-project.